src/python/pants/backend/docker/util_rules/docker_build_context.py

```diff
@@ -166,7 +166,8 @@ def _get_stages_and_tags(
     ) -> tuple[set[str], dict[str, str]]:
         # Go over all FROM tags and names for all stages.
         stage_names: set[str] = set()
-        stage_tags = (tag.split(maxsplit=1) for tag in dockerfile_info.version_tags)
+        # tag is empty if image is referenced by digest instead
+        stage_tags = ((tag.split(maxsplit=1) + [""])[:2] for tag in dockerfile_info.version_tags)
```

Comment:
Is there a more pythonic way?

Reply:
Hmm, I'm not sure about a single-line/"local" refactoring that's more pythonic, maybe this is _very slightly_ neater (using `*` to splat the list into another list, instead of the `+` and extra `()`s that require) 🤷 

```suggestion
stage_tags = ([*tag.split(maxsplit=1), ""][:2] for tag in dockerfile_info.version_tags)
```

Brainstorming a different approach would be to do the splitting within the loop, i.e. have the main iterate `Dockerfile_info.version_tags` directly, e.g.:

```python
for idx, stage_and_tag in enumerate(dockerfile_info.version_tags):
    splits = stage_and_tag.split(maxsplit=1)
    stage = splits[0]
    tag = splits if len(splits) == 2 else None
    if tag is not None and tag.startswith("build-arg:"):
        ...
    ...
```

I don't think either of those are so much better than we need to worry about changing: totally up to you! 😄

Reply:
Like your first approach better. Applied, thanks.